include/swift/AST/Module.h

```diff
@@ -1149,6 +1149,12 @@ class SourceFile final : public FileUnit {
     getInterfaceHash(str);
     out << str << '\n';
   }
+
+  /// Outputs Unified Symbol Resolutions (USRs) and a newline for each
+  /// function declaration in this source file that matches what XCTest
+  /// considers to be a viable test: a class instance method that takes no
+  /// parameters, returns void, and begins with "test".
+  void dumpXCTestMethods(llvm::raw_ostream &out);
```

Comment:
From an API standpoint, I'd prefer that this method gather the XCTest methods into a SmallVector and return that, then the client can render the XCTest method to a file (or whatever is appropriate).